verification/spec/connection-handshake/Environment.tla

```diff
@@ -0,0 +1,233 @@
+---------------------------- MODULE Environment ----------------------------
+
+EXTENDS Naturals, FiniteSets, Sequences
+
+
+CONSTANT MaxHeight,     \* Maximum height of any chain in the system.
+         MaxBufLen      \* Length (size) of message buffers.
+
+ASSUME MaxHeight > 1
+ASSUME MaxBufLen > 1
+
+
+VARIABLES
+    bufChainA,      \* A buffer (sequence) for messages inbound to chain A.
+    bufChainB,      \* A buffer for messages inbound to chain B.
+    storeChainA,    \* The local store of chain A.
+    storeChainB,    \* The local store of chain B.
+    maliciousEnv    \* If TRUE, environment interferes w/ CH protocol. 
+
+
+chainAParameters == [
+    connectionID |-> { "connAtoB" },
+    clientID |-> { "clientChainA" }
+]
+
+chainBParameters == [
+    connectionID |-> { "connBtoA" },
+    clientID |-> { "clientChainB" }
+]
+
+
+ClientIDs == { chainAParameters.clientID, chainBParameters.clientID }
+ConnectionIDs == { chainAParameters.connectionID, chainBParameters.connectionID }
+
+chainAVars == <<bufChainA, bufChainB, storeChainA>>
+chainBVars == <<bufChainA, bufChainB, storeChainB>>
+chainStoreVars == <<storeChainA, storeChainB>>
+allVars == <<chainStoreVars, bufChainA, bufChainB, maliciousEnv>>
+
+
+chmA == INSTANCE ConnectionHandshakeModule
+        WITH MaxHeight      <- MaxHeight,
+             inBuf          <- bufChainA,
+             outBuf         <- bufChainB,
+             store          <- storeChainA,
+             ConnectionIDs  <- chainAParameters.connectionID,
+             ClientIDs      <- chainAParameters.clientID
+
+
+chmB == INSTANCE ConnectionHandshakeModule
+        WITH MaxHeight      <- MaxHeight,
+             inBuf          <- bufChainB,      \* Flip the message buffers w.r.t. chain A buffers.
+             outBuf         <- bufChainA,      \* Inbound for "A" is outbound for "B".
+             store          <- storeChainB,
+             ConnectionIDs  <- chainBParameters.connectionID,
+             ClientIDs      <- chainBParameters.clientID
+
+
+ConnectionStates == {"UNINIT", "INIT", "TRYOPEN", "OPEN"}
+
+
+ConnectionParameters ==
+    [
+        localEnd : chmA!ConnectionEnds,
+        remoteEnd : chmB!ConnectionEnds
+    ]
+
+Connections ==
+    [
+        parameters : ConnectionParameters,
+        state : ConnectionStates
+    ]
+
+Proofs ==
+    [
+        height : 1..MaxHeight
```

Comment:
We should explain what is the semantic we are modelling. As this is simplified view, we need to explain what property of proof (checking) we want to cover.